test/tests/setcomp.py

```diff
@@ -0,0 +1,75 @@
+# this file is adapted from dictcomp.py
+# TODO: this shouldn't be necessary, but our set implementation sucks & doesn't keep things in order
```

Comment:
Not true -- maybe it sucks, but the lack of ordering isn't why :P  Python sets and dicts are well-specified to be hash tables and to have undefined iteration order:
https://docs.python.org/2/library/stdtypes.html#types-set

I think it's still somewhat common for code to rely on set/dict ordering, but I think there's a decent consensus that that's a bug on the user's part.